addons/editor-theme3/theme3.js

```diff
@@ -597,12 +604,18 @@ export default async function ({ addon, console, msg }) {
     Blockly.FieldTextInput.prototype.init = function () {
       // Text inputs
       oldFieldTextInputInit.call(this);
-      if (this.sourceBlock_.isShadow()) return;
-      // Labels in custom block editor
-      this.box_.setAttribute(
-        "fill",
-        isColoredTextMode() ? fieldBackground(this) : this.sourceBlock_.getColourTertiary()
-      );
+      const updateBox_ = () => {
```

Comment:
A comment explaining that this function is needed for compatibility with `recolor-custom-blocks` would be helpful here.